specs/protocol/tx_format.md

```diff
@@ -230,13 +233,40 @@ Note: when verifying a predicate, `txID`, `outputIndex`, `balanceRoot`, `stateRo
 
 Note: when executing a script, `txID`, `outputIndex`, `balanceRoot`, and `stateRoot` are initialized to the transaction ID, output index, amount, and state root of the contract with ID `contractID`, and `txPointer` is initialized to zero.
 
+### InputMessage
+
+| name                  | type        | description                                                            |
+|-----------------------|-------------|------------------------------------------------------------------------|
+| `messageID`           | `byte[32]`  | The messageID as described [here](./identifiers.md#message-id).        |
+| `sender`              | `byte[32]`  | The address of the message sender.                                     |
+| `recipient`           | `byte[32]`  | The address of the message recipient.                                  |
+| `dataLength`          | `uint16`    | Length of message data, in bytes.                                      |
+| `amount`              | `uint64`    | Amount of base asset coins sent with message.                          |
+| `nonce`               | `byte[32]`  | The message nonce.                                                     |
+| `owner`               | `byte[32]`  | Owning address or predicate hash.                                      |
+| `witnessIndex`        | `uint8`     | Index of witness that authorizes spending the coin.                    |
+| `predicateLength`     | `uint16`    | Length of predicate, in instructions.                                  |
+| `predicateDataLength` | `uint16`    | Length of predicate input data, in bytes.                              |
+| `data`                | `byte[]`    | The message data.                                                      |
+| `predicate`           | `byte[]`    | Predicate bytecode.                                                    |
+| `predicateData`       | `byte[]`    | Predicate input data (parameters).                                     |
+
+Transaction is invalid if:
```

Comment:
Hmmm. There should be a bound here on the data length. Maybe make a new consensus parameter, `MAX_MESSAGE_DATA_LENGTH`.